### PR TITLE
Backport of #10746 to v8

### PR DESCRIPTION
**Pull request by fheinecke**
Backport of #10746 into v8. This adds the new APT build tool and APT repository.

### CODE REVIEW

**Comment by fheinecke** [comment]
By the way was dronegen never backported to v8?

**Comment by webvictim** [comment]
@fheinecke It's possible that it wasn't, although I thought it was written before then...

**Comment by wadells** [comment]
> By the way was dronegen never backported to v8?

Dronegen was intentionally removed from v8 in https://github.com/gravitational/teleport/commit/fadcdaf6ea244c7e4e79964666db187543319bf5 https://github.com/gravitational/teleport/pull/9464. If we fix the centos rpm publishing logic, it could be reintroduced.  IMO this probably isn't worth the trouble, since we'll no longer support `v8` once `v11` comes out (probably September or October 2022)

**Review by tcsc — APPROVED** [review]

File: rfd/0058-package-distribution.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+---
```
Comment (wadells) [inline rfd/0058-package-distribution.md:1]: Same remark as https://github.com/gravitational/teleport/pull/13197#pullrequestreview-1001925509, we probably don't need to backport the RFD itself.

**Review by wadells — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]
Technically dronegen **was** in v8 and later removed. But this is fine.

**Pull request merged**